python/scrutiny/server/device/request_dispatcher.py

```diff
@@ -1,59 +1,130 @@
 #    request_dispatcher.py
-#        Use a PriorityQueue to dispatch Request to the device. Associate each request wil
+#        Use a PriorityQueue to dispatch Request to the device. Associate each request with
 #        its callback
 #
 #   - License : MIT - See LICENSE file.
 #   - Project : Scrutiny Debugger (github.com/scrutinydebugger/scrutiny)
 #
 #   Copyright (c) 2021-2022 scrutinydebugger
 
-import queue
+import bisect
+from scrutiny.server.protocol import Response
+from scrutiny.server.server_tools import Throttler
+from time import time
+import math
+import logging
 
 
-class RequestDispatcher:
+class RequestQueue:
+    """
+    Non-thread-safe Queue with priority.
+    Replace queue.PriorityQueue simply because I don't like that they compare data and don't want to introduce workarounds or 
+    dataclass from Python 3.7 just for not comparing the data when selecting priority.
+    We will have all the flexibility we need with our own minimalist custom class
```

Comment:
From python docs:

If the data elements are not comparable, the data can be wrapped in a class that ignores the data item and only compares the priority number:
```
from dataclasses import dataclass, field
from typing import Any

@dataclass(order=True)
class PrioritizedItem:
    priority: int
    item: Any=field(compare=False)
```

Reply:
I would highly suggest you use known working tools and components before writing your own. maybe add a utility function or two to simplify its use

Reply:
yep, saw that. Python version forced to be 3.7+ just because of a design flaw in the original PriorityQueue. I'm not sure I need this for something as simple as a non thread safe queue.